### PR TITLE
Add soundtouch 2.3.2

### PR DESCRIPTION
**Pull request by amyspark**
Hi all,

This is to upstream the wrap I made of the SoundTouch audio processing library.

See [this GStreamer MR](https://gitlab.freedesktop.org/gstreamer/cerbero/-/merge_requests/1404) for further context.

### CODE REVIEW

**Comment by amyspark** [comment]
cc @nirbheek

File: subprojects/packagefiles/soundtouch/meson.build [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    'soundtouch',
+    'c', 'cpp',
+    version: '2.3.2',
+    license: 'LGPL-2.1-or-later',
+    default_options: [
+        'debug=true',
+        'optimization=3',
+        'c_std=c11',
+        # to avoid Meson's warning under MSVC
+        'cpp_std=c++14',
```
Comment (neheb) [inline subprojects/packagefiles/soundtouch/meson.build:11]: https://codeberg.org/soundtouch/soundtouch/commit/5e624fff737db6905ec049d002c583b030ce806d

Comment (amyspark) [reply subprojects/packagefiles/soundtouch/meson.build:11]: This wrap targets 2.3.2: https://codeberg.org/soundtouch/soundtouch/src/tag/2.3.2/CMakeLists.txt#L1-L4. Do you mean to add the flag for future proofing reasons?

Comment (neheb) [reply subprojects/packagefiles/soundtouch/meson.build:11]: Yeah I saw that after. It's fine.

**Comment by amyspark** [comment]
One moment while I run the updates through GStreamer/Cerbero's CI.

**Pull request merged**